fuel-core/src/schema/tx.rs

```diff
@@ -42,18 +42,41 @@ impl TxQuery {
         let db = ctx.data_unchecked::<Database>();
         let key = id.0;
 
-        let tx_pool = ctx.data::<Arc<TxPool>>().unwrap();
+        let tx_status = db.get_tx_status(&key).unwrap().unwrap();
```

Comment:
Going back to the original context in [#152](https://github.com/FuelLabs/fuel-core/issues/152#issuecomment-1076666461), here's what I actually requested:

> - Update query transaction graphql schema to fallback to the txpool find by id method if the tx doesn't exist in the database
> - Update transaction status API to fallback to the txpool if the status doesn't exist in the database, using the Submitted status

Which doesn't quite to map to what you've interpreted here:
> It addresses @Voxelot request for the transaction endpoint to use transaction status an indicator of whether or not to check the mempool or db for a transaction

The difference was that finding the transactions vs fetching the status, were independent and separate things in my mind. We gain no additional information by querying the transaction status first, and functionally nothing would change if we took out this transaction status check. Rather than doubling our I/O per tx request (status column + tx column) we could skip disk I/O entirely in the best case if the tx is still in the pool because we know the status will be submitted.

Also, this wouldn't work at all once we stop storing the `Submitted` status in the db at all, and just deriving that fact from whether the tx exists in the txpool or not.

Reply:
My bad, in my mind I had swapped `transaction api` and `transaction status api` thinking you meant check transaction status as an indicator of whether or not to query the txpool vs the database to find a tx. Will revert the changes I've made then in favor of the correct fix.

To make sure future changes are targeting the right area, can you clarify where the `transaction status api` is (I believe you mean (this function)[https://github.com/FuelLabs/fuel-core/blob/master/fuel-client/src/client.rs#L163] in fuel client? And to make sure I am understanding the logic for this new fix the idea is that when fetching a transaction's status it can be checked if the transaction is in the txpool, and if it is just return `Submitted` which saves a db check?

Reply:
I see how that could be confusing, and may partially be an artifact of the GraphQL architecture vs a straight-forward rpc pattern to some extent. The status is actually a field on the transaction object, which is lazily resolved in this function here: https://github.com/FuelLabs/fuel-core/blob/master/fuel-core/src/schema/tx/types.rs#L215

> when fetching a transaction's status it can be checked if the transaction is in the txpool, and if it is just return Submitted which saves a db check

This is the intention yes. The more I think about it, we may want to even remove the `Submitted` variant from the status enum that's used in the db, and only have it on the graphql schema related types.